### PR TITLE
solution

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_add-todo-form/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
+              onChange={(event) => {
+                setChosenUserId(+event.target.value);
+              }}
+            >
+              {usersFrom.map(user => (
+                <option value={user.id}>
+                  {user.name}
+                </option>
+              ))}
+            </select>
```
Comment (RomekIvantsiv) [inline src/App.tsx:71]: add an empty option `Choose a user` to the users select

File: src/App.tsx [hunk 2]
```diff
@@ -1,17 +1,122 @@
...
+
+  }));
+};
+
+export const preparedTodos = prepareTodos(usersFrom, todosFrom);
 
 const App: React.FC = () => {
+  const [preparedUsers, setPreparedUsers] = useState(preparedTodos);
+  const [inputedTitle, setInputedTitle] = useState('');
+  const [createdId, setCreatedId] = useState(0);
```
Comment (Denchepornyuk) [inline src/App.tsx:25]: you don't need createdId, you can remove it

File: src/App.tsx [hunk 3]
```diff
@@ -1,17 +1,122 @@
...
+      setTitleError('Please enter the title');
+    }
+
+    if (chosenUserId === 0) {
+      setSelectError('Please choose a user');
+    } else {
+      setCreatedId(preparedUsers.length + 1);
+      const createdUser = {
+        title: inputedTitle,
+        id: createdId,
```
Comment (Denchepornyuk) [inline src/App.tsx:41]:
```suggestion
      const createdUser = {
        title: inputedTitle,
        id: preparedUsers.length + 1,
```

File: src/App.tsx [hunk 4]
```diff
@@ -1,17 +1,122 @@
...
-        <span>Users: </span>
-        {users.length}
-      </p>
+            <span className="form__input-error">{selectError}</span>
+            <button
+              type="button"
+              className="btn"
+              onClick={() => {
+                addUser();
+              }}
```
Comment (Denchepornyuk) [inline src/App.tsx:110]:
```suggestion
              onClick={addUser}
```

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]
Screenshot 2022-06-09 at 13 28 34

clear user in select after submitting the form

**Review by Denchepornyuk — APPROVED** [review]

**Pull request closed without merging**